scripts/cat/cats.py

```diff
@@ -0,0 +1,940 @@
+        'smart tactician', 'valuable tactician','valuable insight',
+        'good mediator', 'great mediator', 'excellent mediator',
+        'good teacher', 'great teacher', 'fantastic teacher',
+        'strong connection to StarClan', 'smart', 'very smart', 'extremely smart',
+        'good kitsitter', 'great kitsitter', 'excellent kitsitter', 'camp keeper', 'den builder',
+    ]
+
+    all_cats = {}  # ID: object
+    other_cats = {}  # cats outside the clan
+
+    def __init__(self,
+                 prefix=None,
+                 gender=None,
+                 status="kitten",
+                 parent1=None,
+                 parent2=None,
+                 pelt=None,
+                 eye_colour=None,
+                 suffix=None,
+                 ID=None,
+                 moons=None,
+                 example=False):
+        self.gender = gender
+        self.status = status
+        self.age = None
+        self.skill = None
+        self.trait = None
+        self.parent1 = parent1
+        self.parent2 = parent2
+        self.pelt = pelt
+        self.eye_colour = eye_colour
+        self.mentor = None
+        self.former_mentor = []
+        self.apprentice = []
+        self.former_apprentices = []
+        self.relationships = []
+        self.mate = None
+        self.placement = None
+        self.example = example
+        self.dead = False
+        self.died_by = None  # once the cat dies, tell the cause
+        self.dead_for = 0  # moons
+        self.thought = ''
+        self.genderalign = None
+        self.tortiebase = None
+        self.pattern = None
+        self.tortiepattern = None
+        self.tortiecolour = None
+        self.white_patches = None
+        self.accessory = None
+        self.birth_cooldown = 0
+        self.siblings = []
+
+        # setting ID
+        if ID is None:
+            potential_ID = str(randint(10000, 9999999))
+            while potential_ID in self.all_cats:
+                potential_ID = str(randint(10000, 9999999))
+            self.ID = potential_ID
+        else:
+            self.ID = ID
+
+        # age and status
+        if status is None and moons is None:
+            self.age = choice(self.ages)
+        elif moons != None:
+            for key_age in self.age_moons.keys():
+                if moons in range(self.age_moons[key_age][0], self.age_moons[key_age][1]+1):
+                    self.age = key_age
+                    self.moons = moons
+        else:
+            if status in ['kitten', 'elder']:
+                self.age = status
+            elif status == 'apprentice':
+                self.age = 'adolescent'
+            elif status == 'medicine cat apprentice':
+                self.age = 'adolescent'
+            else:
+                self.age = choice(['young adult', 'adult', 'adult', 'senior adult'])
+        self.moons = random.uniform(self.age_moons[self.age][0], self.age_moons[self.age][1])
```

Comment:
The uniform function returns a float, which results in decimal moons. These should either be rounded before shown to the user or not decimals in the first place (see random.randint())